code/modules/antagonists/revolution/revolution.dm

```diff
@@ -441,56 +438,108 @@
-				continue
-			charter_given = TRUE
-			podspawn(list(
-				"target" = get_turf(headrev_mind.current),
-				"style" = STYLE_SYNDICATE,
-				"spawn" = /obj/item/station_charter/revolution,
-			))
-			to_chat(headrev_mind.current, span_hear("You hear something crackle in your ears for a moment before a voice speaks. \
-				\"Please stand by for a message from your benefactor. Message as follows, provocateur. \
-				<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\""))
-		SEND_GLOBAL_SIGNAL(COMSIG_GLOB_REVOLUTION_TAX_REMOVAL)
-
-		for (var/mob/living/player as anything in GLOB.player_list)
-			var/datum/mind/player_mind = player.mind
-
-			if (isnull(player_mind))
-				continue
+		victory_effects()
 
-			if (!(player_mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
-				continue
+/datum/team/revolution/proc/victory_effects(revs_win_injection_amount)
+	var/charter_given = FALSE
 
-			if (player_mind in ex_revs + ex_headrevs)
-				continue
+	for(var/datum/mind/headrev_mind as anything in ex_headrevs)
+		if(charter_given)
+			break
+		if(!headrev_mind.current || headrev_mind.current.stat != CONSCIOUS)
+			continue
+		charter_given = TRUE
+		podspawn(list(
+			"target" = get_turf(headrev_mind.current),
+			"style" = STYLE_SYNDICATE,
+			"spawn" = /obj/item/station_charter/revolution,
+		))
+		to_chat(headrev_mind.current, span_hear("You hear something crackle in your ears for a moment before a voice speaks. \
+			\"Please stand by for a message from your benefactor. Message as follows, provocateur. \
+			<b>You have been chosen out of your fellow provocateurs to rename the station. Choose wisely.</b> Message ends.\""))
+	SEND_GLOBAL_SIGNAL(COMSIG_GLOB_REVOLUTION_TAX_REMOVAL)
 
-			player_mind.add_antag_datum(/datum/antagonist/enemy_of_the_revolution)
+	for (var/mob/living/player as anything in GLOB.player_list)
+		var/datum/mind/player_mind = player.mind
 
-			if (!istype(player))
-				continue
+		if (isnull(player_mind))
+			continue
 
-			if(player_mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
-				ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, REF(src))
-				player.med_hud_set_status()
+		if (!(player_mind.assigned_role.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
+			continue
 
-		for(var/datum/job/job as anything in SSjob.joinable_occupations)
-			if(!(job.departments_bitflags & (DEPARTMENT_BITFLAG_SECURITY|DEPARTMENT_BITFLAG_COMMAND)))
-				continue
-			job.allow_bureaucratic_error = FALSE
-			job.total_positions = 0
+		if (player_mind in ex_revs + ex_headrevs)
+			continue
+
+		player_mind.add_antag_datum(/datum/antagonist/enemy_of_the_revolution)
+
+		if (!istype(player))
+			continue
+
+		if(player_mind.assigned_role.departments_bitflags & DEPARTMENT_BITFLAG_COMMAND)
+			ADD_TRAIT(player, TRAIT_DEFIB_BLACKLISTED, REF(src))
```

Comment:
god I wish we could lint for this done improperly

Reply:
yeah

Reply:
we can absolutely lint for this with dm-lua

Reply:
so 2026